dask-gateway-server/dask_gateway_server/backends/kubernetes/controller.py

```diff
@@ -1175,6 +1175,10 @@ def make_pod(self, namespace, cluster_name, config, is_worker=False):
             },
         }
 
+        pod["spec"]["imagePullSecrets"] = []
+        for image_pull_secret in config.image_pull_secrets:
+            pod["spec"]["imagePullSecrets"].append({"name": image_pull_secret})
+
```

Comment:
- [ ] Make this conditional, so that providing an empty list doesn't make the pod specification cluttered?
- [ ] Make this a straight passthrough of k8s native specification of imagePullSecrets, don't make it accept the names of k8s secrets but accept a list of `{"name": "secret-name-here"}`.

  I'd like that as it aligns with the helm chart's other imagePullSecrets specification. See for example:
  https://github.com/dask/dask-gateway/blob/a660e2a4820a51e4027294bbd6ff8ed17e477563/resources/helm/dask-gateway/templates/gateway/deployment.yaml#L37-L40

Reply:
```suggestion
        if config.image_pull_secrets:
            pod["spec"]["imagePullSecrets"] = config.image_pull_secrets

```